packages/test-utils/src/reporter.ts

```diff
@@ -530,16 +539,15 @@ class ReplayReporter<TRecordingMetadata extends UnstructuredMetadata = Unstructu
     }
   }
 
-  private async _addTestsToShard(tests: TestRunTestInputModel[]): Promise<TestRunTestsPendingWork> {
+  private async _addTestsToShard(
+    tests: TestRunTestInputModel[]
+  ): Promise<TestRunTestsPendingWork | undefined> {
     let testRunShardId = this._testRunShardId;
     if (!testRunShardId) {
       await this._testRunShardIdPromise;
       testRunShardId = this._testRunShardId;
       if (!testRunShardId) {
-        return {
-          type: "test-run-tests",
-          error: new Error("Unable to add tests to test run: ID not set"),
-        };
+        return;
```

Comment:
returning `undefined` here helps us to avoid the error propagation and a situation like this:
```
[replay.io]: 🕑 Completing some outstanding work ...
[replay.io]: 
[replay.io]: ❌ We encountered some unexpected errors creating your tests on replay.io
[replay.io]:    - GraphQL request for CreateTestRunShard failed (Unknown API key)
[replay.io]:    - Unable to complete test run: ID not set
```

If the test run shard couldn't be created we only need to report that error. Everything that fails because of that is just its consequence and noise